woocommerce/rest-api/Controllers/Settings.php

```diff
@@ -125,7 +125,35 @@ public function get_items_permissions_check( $request ) {
 	// TODO: get_items()
 
 
-	// TODO: get_item()
+	/**
+	 * Gets a single setting.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param \WP_REST_Request $request request object
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function get_item( $request ) {
+
+		$setting_id = $request->get_param( 'id' );
+
+		if ( $setting = $this->settings->get_setting( $setting_id ) ) {
+
+			return $this->prepare_item_for_response( $setting, $request );
+
+		} else {
+
+			return new \WP_Error(
+				'wc_rest_setting_not_found',
+				sprintf(
+					/* translators: Placeholder: %s - setting ID */
+					__( 'Setting %s does not exist', 'woocommerce-plugin-framework' ),
+					$setting_id
+				),
+				[ 'status' => 404 ]
```

Comment:
Is this how we want to set the status? Will it translate to REST status? I am still trying to make the API work in Postman so I can check.

Reply:
Yes, I think that should work. See [`class-wp-rest-server.php#L159-L165`](https://github.com/WordPress/WordPress/blob/55482506a3cba25dbef5930d25b09611029c3264/wp-includes/rest-api/class-wp-rest-server.php#L159-L165).